bionic/task_state.py

```diff
@@ -77,6 +77,17 @@ def is_completable(self):
         """
         return self._is_initialized and len(self.blocking_dep_states()) == 0
 
+    # TODO Now that we've introduced this method, let's remove
+    # provider.attrs.should_persist(), which is just a wrapper for
+    # provider.attrs._can_persist anyway.
+    def should_persist(self):
+        return self.provider.attrs.should_persist() and not self.has_missing_output()
+
+    def has_missing_output(self):
```

Comment:
It's not very clear what missing output means here. Maybe `missing_dep_outputs`?

Reply:
This actually means that the *output* of the task has (or would have) a missing value. Maybe `output_would_be_missing`?